components/link_checker/src/lib.rs

```diff
@@ -70,9 +81,32 @@ pub fn check_url(url: &str) -> LinkResult {
     res
 }
 
+fn has_anchor(url: &str) -> bool {
+    match url.find('#') {
+        Some(index) => match url.get(index..=index + 1) {
+            Some("#/") | Some("#!") | None => false,
+            Some(_) => true,
+        },
+        None => false,
+    }
+}
+
+fn check_page_for_anchor(url: &str, body: reqwest::Result<String>) -> Result<()> {
+    let body = body.unwrap();
+    let index = url.find('#').unwrap();
+    let anchor = url.get(index + 1..).unwrap();
+    let checks: [String; 2] = [format!(" id='{}'", anchor), format!(r#" id="{}""#, anchor)];
```

Comment:
I had completely forgotten but anchors can also be `name=` rather than id so there are actually 4 checks

Reply:
I've pushed it directly to the PR